### PR TITLE
[NO TICKET] test code coverage with symlinked dependency required via entrypoint

### PR DESCRIPTION
**Pull request by anmarchenko**
**What does this PR do?**
This PR adds a test to assert that code coverage tool gets correct file paths when dependencies are located in symlinked folder and required via single entrypoint (emulating usage of local dependencies that are just symlinks to modules in minirepo).

**Motivation**
Make sure that code coverage won't break in the future as this is known case for the monorepo setup.

**How to test the change?**
This PR adds only unit tests.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -103,7 +103,7 @@ namespace :test do
           "bundle exec appraisal #{ruby_runtime}-#{appraisal_group} rake #{spec_task}"
```
Comment (anmarchenko) [inline Rakefile:103]: changes in this file are unrelated, just enjoying our own Datadog's static code analysis tool

**Review by nikita-tkachenko-datadog — APPROVED** [review]
